extensions/resteasy-reactive/quarkus-resteasy-reactive-jackson/runtime/src/main/java/io/quarkus/resteasy/reactive/jackson/runtime/serialisers/JacksonMessageBodyReader.java

```diff
@@ -56,4 +56,15 @@ private Object doReadFrom(Class<Object> type, Type genericType, InputStream enti
         return reader.forType(reader.getTypeFactory().constructType(genericType != null ? genericType : type))
                 .readValue(entityStream);
     }
+
+    private boolean isReadable(MediaType mediaType, Class<?> type, Type genericType) {
+        if (mediaType == null) {
+            return false;
+        }
+        if (String.class.equals(type) && (String.class.equals(genericType) || (genericType == null))) { // don't attempt to read plain strings
```

Comment:
This is confusing given that `String` is not generic, so if `type == String.class`  then `genericType == String.class` must also be true or else something is super wrong in RR.

Reply:
The idea here is to differentiate between `String` and `List<String>`. In the latter case, `genericType` will be a `ParameterizedTypeImpl` which we assume Jackson can handle.

A simple `String`, we know Jackson can't handle.

Reply:
But if `genericType` is `List<String>` then `type == List.class`, not `String.class`.

Reply:
Yeah, which is why the method returns `false`, no?

Reply:
My point is that if `String.class.equals(type)` then the second part of the condition must ALWAYS be true. So it's redundant.

Reply:
OK, I see your point

Reply:
Checked this again and now I see why I did what I did.

When `genericType` is `List<String>`, then `type` is `String`. This feels very wrong, but I am not sure I want to tackle it... (see my other comment).
I'll make a lame attempt to fix it, but if things start to break, I really prefer to leave it as is for now and fix later properly on when we are all refreshed :)

Reply:
Ah, so that's another consequence of the other issue. I can try to find the underlying issue tomorrow, I'd prefer to at least know what it is before adding weird stuff like this.

Reply:
Yeah. I'll look into it more

Reply:
I'll let you know what I find, but indeed the last commit might have to be changed considerably.

Reply:
I see where I messed up - looking to see what can be done.

Thanks for insisting :)

Reply:
Fixed, properly. 

Thanks again for being vigilant